component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -193,9 +197,24 @@ protected void enrichCustomClaims(Map<String, Object> customClaims, IdentityProv
      * @return whether the audience is valid or not
      */
     protected boolean validateAudience(List<String> audiences, IdentityProvider idp, String requestedAudience,
-                                       RequestParameter[] params) {
+                                       RequestParameter[] params, String tenantDomain) throws IdentityOAuth2Exception {
 
-        return audiences != null && audiences.stream().anyMatch(aud -> aud.equals(idp.getAlias()));
+        String idpIssuerName = OAuth2Util.getIssuerLocation(tenantDomain);
+        boolean audienceFound = audiences != null && audiences.contains(idpIssuerName);
+
+        if (audienceFound) {
+            return true;
+        }
+
+        String idpAlias = getIDPAlias(idp, tenantDomain);
+        if (StringUtils.isEmpty(idpAlias)) {
+            handleException(OAuth2ErrorCodes.INVALID_REQUEST, "Issuer name of the token issuer is not " +
+                    "included as a audience. Alias of the local Identity Provider has not "
+                    + "been configured for " + idp.getIdentityProviderName());
+        } else {
```

Comment:
We can replace

else {
            return audiences != null && audiences.stream().anyMatch(aud -> aud.equals(idpAlias));
        }
        return false;
        
        
 with return audiences != null && audiences.stream().anyMatch(aud -> aud.equals(idpAlias));